.github/workflows/publish.yml

```diff
@@ -50,6 +52,10 @@ jobs:
           python -m pip install --upgrade pip
           pip install -e ".[dev, docs]"
           sudo apt-get install graphviz
+      - name: Checkout latest tagged commit
+        run: |
+          git checkout $(git tag -l | tail -1)
+          tox -e build
```

Comment:
I suppose this step is needed only in the case when the publish job is triggered manually, right? If it is triggered automatically, then it is triggered because a new tag/release was created and the last commit is tagged, so `fetch-depth: 0` is enough.

Reply:
I am thinking, do we really need this? What are the use cases of triggering the job manually. Maybe it need not be truncated to the last tagged commit?

Reply:
I don't know about the need of it, but `workflow_dispatch` is there, so manual triggering is possible, and the checkout step is there to handle such usage.

Reply:
Yeah, I guess there is no harm. The only situation I can think of is the situation that the automatic run fails and someone needs to rerun it manually.

Reply:
The checkout step is only needed in `publish_docs` job.  Code builds are done on the latest commit, tagged or not.

Reply:
BTW, why is `tox -e- build` part of this step? The actual building of the docs is done in the next step with `tox -e docs`

Reply:
Hmm, I am not sure how it helps actually. I just did a quick testing locally and it does not seem to help at all. We are using `setuptools-scm` for guessing the version from git metadata. Maybe this tools has some options to handle this situation more robustly?